File: frontend/src/models/TreeholeSearchResult.ts
import { SearchSourceInfo } from "./SearchResult";

export interface TreeholeSearchResult {
  results: TreeholeSearchResultItem[];
  sourceInfo: SearchSourceInfo;
  isSuccess: boolean;
  errorMessage: string | null;
}

export interface TreeholeSearchResultItem {
  pid: number;
  text: string;
  time: string;
  imageId: number | null;
  commentCount: number;
  starCount: number;
}

export interface TreeholeComment {
  cid: number;
  pid: number;
  text: string;
  timestamp: string;
  tag: any;
  comment_id: any;
  name: string;
  quote: TreeholeCommentQuoteData | null;
}

export interface TreeholeCommentQuoteData {
  pid: number;
  name_tag: string;
  text: string;
}

File: PkuAggregated/Models/TreeholeSearchResult.cs
using PkuAggregated.Interfaces;

namespace PkuAggregated.Models
{
    public class TreeholeSearchResult
    {
        public required List<TreeholeSearchResultItem> Results { get; set; }
        public required SearchSourceInfo SourceInfo { get; set; }
        public required bool IsSuccess { get; set; }
        public string? ErrorMessage { get; set; }
    }

    public class TreeholeSearchResultItem
    {
        public required int Pid { get; set; }
        public required string Text { get; set; }
        public required DateTime Time { get; set; }
        public int? ImageId { get; set; }
        public int CommentCount { get; set; }
        public int StartCount { get; set; }
    }
}
